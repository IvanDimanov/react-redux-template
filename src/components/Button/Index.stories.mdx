import {Meta, Preview, Story, Props} from '@storybook/addon-docs/blocks';

import Button from '../Button';

<Meta
  title='Design System/Components/Button'
  component={Button}
/>


# Component/Button
A basic UI button component that allows to execute an action in one click.<br />
This component is based on the [Material UI Button Component](https://material-ui.com/components/buttons/).


#### Default
<Preview>
  <Story name='Default'>
    <Button>Default Button</Button>
  </Story>
</Preview>


#### Secondary
<Preview>
  <Story name='Secondary'>
    <Button color='secondary'>Secondary colors Button</Button>
  </Story>
</Preview>


#### Loading and Disabled
<Preview>
  <Story name='Loading'>
    <Button isLoading disabled />
  </Story>
</Preview>
