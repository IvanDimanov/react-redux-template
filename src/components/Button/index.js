import React, {memo} from 'react'
import PropTypes from 'prop-types'

import MuiButton from '@material-ui/core/Button'

import {Spinner} from 'components'


const Button = ({children, isLoading, ...rest}) => (
  <MuiButton {...rest}>
    {isLoading && <Spinner size={50} />}
    {children}
  </MuiButton>
)


Button.propTypes = {
  children: PropTypes.node,
  isLoading: PropTypes.bool,
}

Button.defaultProps = {
  children: 'Button',
  isLoading: false,
}

Button.displayName = 'Button'


export default memo(Button)
