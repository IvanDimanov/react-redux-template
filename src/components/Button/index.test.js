import React from 'react'

import {shallow} from 'enzyme'

import Button from './index'

describe('Button', () => {
  describe('Default view', () => {
    let wrapper
    beforeEach(() => {
      wrapper = shallow(<Button />)
    })

    it('should match the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
