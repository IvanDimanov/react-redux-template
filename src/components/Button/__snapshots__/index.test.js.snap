// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Default view should match the snapshot 1`] = `"<button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Button</span></button>"`;
